core/src/main/java/org/infinispan/factories/ComponentRegistry.java

```diff
@@ -217,6 +226,37 @@ public void start() {
       super.start();
    }
 
+   /**
+    * The {@link ComponentRegistry} may delay the start after a cluster shutdown. The component waits for
+    * the installation of the previous stable topology to start.
+    *
+    * Caches that do not need state transfer or are private do not need to delay the start.
+    */
+   @Override
+   protected CompletionStage<Void> delayStart() {
```

Comment:
While this is okay... I am not liking where this code is residing. This is putting too much logic in the component registry itself. The registry should be more about registering components, not knowing inner details of them.

I wonder is it possible to move this logic to an appropriate interceptor or component? For example we can add the clear logic to the persistence manager and stable topology directly to the LocalTopologyManager/StateTransferManager. Then we would need to disconnect start method invocation from the actual interceptor initialization and leave the state as INITIALIZING until it is done. I _think_ this would keep the logic more closely aligned with where it should be.